ipatests/test_integration/test_caless.py

```diff
@@ -306,21 +303,53 @@ def get_replica_filename(self, replica):
                             'replica-info.gpg')
 
     @classmethod
-    def export_pkcs12(cls, nickname, filename='server.p12', password=None):
-        """Export a cert as PKCS#12 to the given file"""
+    def create_pkcs12(cls, nickname, filename='server.p12', password=None):
+        """Create a cert chain and generate pkcs12 cert"""
         if password is None:
             password = cls.cert_password
-        ipautil.run(['pk12util',
-                     '-o', filename,
-                     '-n', nickname,
-                     '-d', 'nssdb',
-                     '-K', cls.cert_password,
-                     '-W', password], cwd=cls.cert_dir)
+
+        fname_chain = []
+
+        key_fname = os.path.join(cls.cert_dir, nickname + '.key')
+        certchain_fname = os.path.join(cls.cert_dir, nickname + '.pem')
+
+        nick_chain = nickname.split(os.sep)
+
+        # to construct whole chain e.g "ca1 - ca/sub - ca1/sub/server"
+        for index, _value in enumerate(nick_chain):
+            cert_path = '/'.join(nick_chain[:index+1])
+            fname_chain.append(os.path.join(cls.cert_dir, cert_path + '.crt'))
+        # to simulate unknown CA we drop ca3 from the chain
+        if nickname.startswith('ca3/'):
+            fname_chain.pop(0)
+
+        with open(certchain_fname, 'w') as chain:
+                for cert_fname in fname_chain:
+                    with open(cert_fname) as cert:
+                        chain.write(cert.read())
```

Comment:
Instead of having a separate file you can just create the `fname_chain` as a list of arguments to `ipautil.run`. So instead of having the `-in` argument to `openssl`, you extend the argument list by a list which should look somehow like this:
```python
['-certfile', 'ca1.crt', '-certfile', 'ca1/server.crt']
```
and so on.
I am doing the very same in https://github.com/freeipa/freeipa-tools/pull/46/files except that's in bash. You can just rewrite it to python.